app/models/reminder_mailing.rb

```diff
@@ -0,0 +1,19 @@
+class ReminderMailing < ApplicationRecord
+  belongs_to :creator, -> { with_deleted }, class_name: 'User'
+  has_many :reminder_mailing_volunteers, dependent: :destroy
+  accepts_nested_attributes_for :reminder_mailing_volunteers
+
+  has_many :volunteers, through: :reminder_mailing_volunteers
+  enum kind: { probation_period: 1, half_year: 0 }
+
+  # setter generates relation to assignment/group_assignment and volunteer in one go
+  def reminder_mailing_volunteers=(reminder_mailable)
+    if [Assignment, GroupAssignment].include? reminder_mailable.first&.class
+      super(reminder_mailable.map do |mailable|
+        ReminderMailingVolunteer.new(volunteer: mailable.volunteer, reminder_mailable: mailable)
+      end)
+    else
+      super(reminder_mailable)
```

Comment:
I'm a bit confused by this, is there a case where the `reminder_mailables` are **not** assignments or group assignments?

Reply:
ReminderMailing -> ReminderMailingVolunteer -> Assignment or GroupAssignment

But with this setter i can pass:
```ruby
ReminderMailing.new(reminder_mailing_volunteers: [Assignment])
```
so the setter will automaticly build a GroupAssignmentVolunteer record with the relation to the Assignment and Volunteer in one go

Reply:
Otherwise it treats it just as a normal ReminderMailingVolunteer record beeing passed